civic_scraper/platforms/legistar/scraper.py

```diff
@@ -0,0 +1,82 @@
+import civic_scraper
+from civic_scraper import base
+from civic_scraper.base.asset import Asset, AssetCollection
+from civic_scraper.base.cache import Cache
+
+from legistar.events import LegistarEventsScraper
+
+from datetime import datetime
+from pathlib import Path
+from urllib.parse import urlparse, parse_qs
+
+# Scrape today's agendas and minutes from a Legistar site
+class LegistarSite(base.Site):
+
+    def __init__(self, base_url, legistar_instance, state, cache=Cache(), parser_kls=None, timezone=None):
+        super().__init__(base_url, cache, parser_kls)
+        self.legistar_instance = legistar_instance
+        self.state = state
+        self.timezone = timezone
+
+    def create_asset(self, event, scraper):
+        # get date and time of event
+        meeting_datetime = " ".join((event['Meeting Date'], event['Meeting Time']))
+        meeting_date = scraper.toDate(meeting_datetime)
+        meeting_time = scraper.toTime(meeting_datetime)
+
+        # get event ID
+        if type(event['Meeting Details']) == dict:
+            url = event['Meeting Details']['url']
+            query_dict = parse_qs(urlparse(url).query)
+
+            meeting_id = 'legistar_{}-{}_{}'.format(self.state, self.legistar_instance, query_dict['ID'][0])
+        else:
+            # No meeting details, e.g., event is in future
+            url = None
+            meeting_id = None
+
+        # get event name
+        if type(event['Name']) == dict:
+            asset_name = event['Name']['label']
+            committee_name = event['Name']['label']
+        else:
+            asset_name = event['Name']
+            committee_name = event['Name']
+
+        e = {'url': url,
+             'asset_name': asset_name,
+             'committee_name': committee_name,
+             'place': event['Meeting Location'],
+             'state_or_province': self.state,
+             'asset_type': 'Agenda',
+             'meeting_date': meeting_date,
+             'meeting_time': meeting_time,
+             'meeting_id': meeting_id,
+             'scraped_by': f'civic-scraper_{civic_scraper.__version__}',
+             'content_type': 'txt',
+             'content_length': None,
+        }
+        return Asset(**e)
+
+    def scrape(self, download=True):
+        webscraper = LegistarEventsScraper(retry_attempts=3)
+
+        # required to instantiate webscraper
+        webscraper.BASE_URL = "https://{}.legistar.com/".format(self.legistar_instance)
```

Comment:
it would be much safer to use get the base domain name from self.url. here's one way to do that: https://stackoverflow.com/a/56476496/98080